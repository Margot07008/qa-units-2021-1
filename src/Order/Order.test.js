jest.mock('../utils/getDate')
import React from 'react'
import {configure, shallow} from "enzyme";
import Adapter from 'enzyme-adapter-react-16';
import Order from "./Order";
import {getDate} from '../utils/getDate';
import {fakeOrders} from "../data/fakeOrders";
import toJson from "enzyme-to-json";


const fakeDate = 123;
configure({adapter: new Adapter()});


describe('Order.js', () => {

    beforeEach(() => {
        getDate.mockClear();
        getDate.mockReturnValue(fakeDate);
    });

    afterAll(() => {
        getDate.mockReset();
    });

    it('render empty', () => {
        const wrapper = shallow(<Order/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('render without params', () => {
        const wrapper = shallow(<Order order={{}}/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('render with data', () => {
        const wrapper = shallow(<Order order={{shop: "test1", date: 123}}/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('render with data and items', () => {
        const wrapper = shallow(<Order order={{shop: "test2", date: 123, items: ["item1", "item2"]}}/>);
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('check call getDate', () => {
        shallow(<Order order={fakeOrders[0]}/>);
        expect(getDate).toHaveBeenCalled();
    });

});