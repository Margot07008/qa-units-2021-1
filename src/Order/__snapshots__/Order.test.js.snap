// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order.js render empty 1`] = `""`;

exports[`Order.js render with data 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      test1
    </span>
    <span
      id="date"
    >
      123
    </span>
  </div>
  <div
    className="Order-items"
  />
</div>
`;

exports[`Order.js render with data and items 1`] = `
<div
  className="Order"
>
  <div
    className="Order-header"
  >
    <span
      className="Order-shop"
    >
      test2
    </span>
    <span
      id="date"
    >
      123
    </span>
  </div>
  <div
    className="Order-items"
  >
    <div
      className="Order-item"
      key="0"
    >
      item1
    </div>
    <div
      className="Order-item"
      key="1"
    >
      item2
    </div>
  </div>
</div>
`;

exports[`Order.js render without params 1`] = `""`;
